kDrive/AppRouter.swift

```diff
@@ -0,0 +1,763 @@
+    ///   - frozenFile: A frozen file to display
+    ///   - driveFileManager: driveFileManager
+    ///   - navigationController: The navigation controller to use
+    ///   - animated: Should be animated
+    @MainActor func presentFileDetails(
+        frozenFile: File,
+        driveFileManager: DriveFileManager,
+        navigationController: UINavigationController,
+        animated: Bool
+    )
+
+    /// Present the InApp purchase StoreViewController
+    /// - Parameters:
+    ///   - driveFileManager: driveFileManager
+    ///   - navigationController: The navigation controller to use
+    ///   - animated: Should be animated
+    @MainActor func presentStoreViewController(
+        driveFileManager: DriveFileManager,
+        navigationController: UINavigationController,
+        animated: Bool
+    )
+}
+
+/// Something that can set an arbitrary RootView controller
+public protocol RouterRootNavigable {
+    /// Something that can set an arbitrary RootView controller
+    ///
+    /// Should not be used externally except by SceneDelegate.
+    @MainActor func setRootViewController(_ viewController: UIViewController,
+                                          animated: Bool)
+
+    /// Setup the root of the view stack
+    /// - Parameters:
+    ///   - currentState: the state to present
+    ///   - restoration: try to restore scene or not
+    @MainActor func prepareRootViewController(currentState: RootViewControllerState, restoration: Bool)
+
+    /// Set the main theme color
+    @MainActor func updateTheme()
+}
+
+public protocol TopmostViewControllerFetchable {
+    /// Access the current top most ViewController
+    @MainActor var topMostViewController: UIViewController? { get }
+}
+
+/// Actions performed by router, `async` by design
+public protocol RouterActionable {
+    /// Ask the user to review the app
+    func askForReview() async
+
+    /// Ask the user to remove pictures if configured
+    func askUserToRemovePicturesIfNecessary() async
+
+    func refreshCacheScanLibraryAndUpload(preload: Bool, isSwitching: Bool) async
+}
+
+/// Something that can navigate within the kDrive app
+public typealias AppNavigable = RouterActionable
+    & RouterAppNavigable
+    & RouterFileNavigable
+    & RouterRootNavigable
+    & TopmostViewControllerFetchable
+
+public struct AppRouter: AppNavigable {
+    @LazyInjectService private var driveInfosManager: DriveInfosManager
+    @LazyInjectService private var keychainHelper: KeychainHelper
+    @LazyInjectService private var reviewManager: ReviewManageable
+    @LazyInjectService private var availableOfflineManager: AvailableOfflineManageable
+    @LazyInjectService private var backgroundUploadSessionManager: BackgroundUploadSessionManager
+    @LazyInjectService private var accountManager: AccountManageable
+
+    /// Get the current window from the app scene
+    @MainActor private var window: UIWindow? {
+        let scene = UIApplication.shared.connectedScenes.first { scene in
```

Comment:
So we still support only one scene, correct ?

Reply:
You are correct, single for now.